dev/release/04-binary-download.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+set -ex
+
+main() {
+    local -r source_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+    local -r source_top_dir="$( cd "${source_dir}/../../" && pwd )"
+    pushd "${source_top_dir}"
+
+    if [ "$#" -ne 2 ]; then
+        echo "Usage: $0 <version> <rc-num>"
+        exit 1
+    fi
+
+    local -r version="$1"
+    local -r rc_number="$2"
+    local -r tag="adbc-${version}"
+    local -r rc_branch="release-${version}-rc${rc_number}"
+
+    : ${REPOSITORY:="apache/arrow"}
+
+    echo "Waiting for GitHub Actions workflow on ${REPOSITORY}:${rc_branch}"
+
+    local -r run_id=$(gh run list \
+                         --repo "${REPOSITORY}" \
+                         --workflow=packaging-wheels.yml \
+                         --json 'databaseId,event,headBranch' \
+                         --jq ".[] | select(.event == \"workflow_dispatch\" and .headBranch == \"${rc_branch}\") | .databaseId" \
+                          | head -n1)
+
+    echo "Found GitHub Actions workflow with ID: ${run_id}"
+    gh run watch --repo "${REPOSITORY}" --exit-status "${run_id}"
+    gh run view --repo "${REPOSITORY}" "${run_id}"
+
+    local -r download_dir="packages/release-${version}-rc${rc_number}"
+    rm -rf "${download_dir}"
+    mkdir -p "${download_dir}"
+    gh run download --repo "${REPOSITORY}" --dir "${download_dir}" "${run_id}"
```

Comment:
Releases might still be convenient for discoverability, but I suppose then you run into the problem of having to clean them up.

Reply:
How about uploading artifacts to GitHub Releases in `.github/workflows/packaging-wheels.yml` instead of creating this file?
We can use https://github.com/softprops/action-gh-release for it.

We can upload signs and checksums for artifacts in GitHub Releases separately by adding `dev/release/04-binary-sign.sh`. It lists artifacts in GitHub Releases, signs them, computes checksums of them and uploads these signs and checksums.

Reply:
Ok, I'll give that a try!

Reply:
Oh, so to be clear, then when we create the final tag, it'll create a new release and we'll build the artifacts again? Or else, we might have to fiddle with the action so that it only runs on RC branches.

Also, looks like we'll have to ask INFRA to enable the action since it doesn't seem to be one of the pre-approved ones.

Reply:
> we'll build the artifacts again?

No. We must copy the approved artifacts from RC's GitHub Releases. I think that we can implement this by GitHub Actions.

> Also, looks like we'll have to ask INFRA to enable the action since it doesn't seem to be one of the pre-approved ones.

Oh...

Reply:
Right. I'll restrict it to only run on the RC tags, then. 

Worst case, I think we have `gh` in actions so I can just manually script it.

Reply:
Working on it here now: https://github.com/apache/arrow-adbc/pull/208